src/Dependency/Repository/DependencyRepository.php

```diff
@@ -27,53 +37,54 @@
     use ElementProviderTrait;
 
     public function __construct(
-        private ServiceResolverInterface $serviceResolver,
+        private ElementSearchServiceInterface $elementSearchService,
+        private SearchProviderInterface $searchProvider
     ) {
     }
 
-    public function listRequiresDependencies(string $elementType, int $elementId): array
+    public function listRequiresDependencies(
+        ElementParameters $elementParameters,
+        DependencyParameters $parameters,
+        UserInterface $user
+    ): ElementSearchResult
     {
-        return $this->getElement(
-            $this->serviceResolver,
-            $elementType,
-            $elementId
-        )
-            ->getDependencies()
-            ->getRequires();
+        $search = $this->searchProvider->createAssetSearch();
```

Comment:
Just for my information, the data from the SearchProvider is coming from the Search Server, and we treat it as if it were a DB. Is this why it is in a Repository?

Reply:
Yeah kind of. Maybe the name is not fitting, but then I really need to refactor the whole thing and maybe put everything into the service or move it to some other class. 
Not sure thought I just tried to make it work with the GDI